billing/test/lib/egress-traffic-queue.js

```diff
@@ -0,0 +1,68 @@
+
+/** @type {import('./api').TestSuite<import('./api').EgressTestContext>} */
+export const test = {
+    'should process egress events': async (/** @type {import('entail').assert} */ assert, ctx) => {
+        const maxEvents = 100
+        const events = await Promise.all(
+            Array.from({ length: maxEvents }, () => randomEgressEvent())
+        )
+
+        // add egress events to the queue to simulate events from the Freeway worker
+        for (const e of events) {
+            console.log(`Adding egress event to the queue: CustomerId: ${e.customerId}, ResourceId: ${e.resourceId}, Timestamp: ${e.timestamp.toISOString()}`)
+            await ctx.egressQueue.add(e)
+        }
+       
+        // simulate the SQS event that triggers the handler
+        // FIXME (fforbeck): why the events are not collected?
+        const collected = await collectQueueMessages(ctx.egressQueue)
+        assert.ok(collected.ok, 'Failed to collect queue messages')
+        assert.equal(collected.ok.length, events.length, 'Collected queue messages length does not match')
+        
+        // @type {import('aws-lambda').SQSEvent}
+        const sqsEvent = {
+            Records: collected.ok.map(e => ({
+                // @type {import('aws-lambda').SQSRecord}
+                body: JSON.stringify(e),
+                messageId: Math.random().toString(),
+                receiptHandle: Math.random().toString(),
+                awsRegion: ctx.region,
+                eventSource: 'aws:sqs',
+                eventSourceARN: `arn:aws:sqs:${ctx.region}:${ctx.accountId}:${ctx.queueUrl}`,
+                awsAccountId: ctx.accountId,
+                md5OfBody: '',
+                md5OfMessageAttributes: '',
+                attributes: {
+                    ApproximateReceiveCount: '1',
+                    SentTimestamp: e.timestamp.getTime().toString(),
+                    SenderId: ctx.accountId,
+                    ApproximateFirstReceiveTimestamp: e.timestamp.getTime().toString(),
+                },
+                messageAttributes: {},
+            }))
+        }
+
+        const response = await ctx.egressHandler(sqsEvent, ctx)
+        assert.equal(response.statusCode, 200)
+        assert.equal(response.body, 'Egress events processed successfully')
+
+        // ensure we got a egress record for each event
+        for (const e of events) {
+            const record = await ctx.egressEventStore.list({
+                customerId: e.customerId,
+                resourceId: e.resourceId,
+                from: e.timestamp,
+            })
+            assert.ok(record.ok)
+            assert.equal(record.ok.results.length, 1)
+            assert.equal(record.ok.results[0].customerId, e.customerId)
+            assert.equal(record.ok.results[0].resourceId, e.resourceId)
+            assert.equal(record.ok.results[0].timestamp, e.timestamp)
+        }
+        // FIXME (fforbeck): how to check we send the events to stripe?
```

Comment:
@travis, How we can test if the Stripe Usage Record API is being called for each event?

Reply:
goooood question - I think the most sophisticated test logic we have for stripe is here:

https://github.com/storacha/w3infra/blob/main/upload-api/test/billing.test.js

it might be worth trying to use this:

https://github.com/stripe/stripe-mock

but I don't think we've used it for anything yet

it's also worth looking into this:

https://docs.stripe.com/billing/testing

high level though I think our Stripe testing overall can be improved! happy to kick around ideas for this!